### PR TITLE
[riverpod_graph] Add a test with a Consumer widget

### PR DESCRIPTION
**Pull request by ValentinVignal**
In this PR:
- I remove riverpod from the dev decencies
- I created a test that test the consumer widgets are actually displayed in the graph (there was no test with a consumer widget before)

### CODE REVIEW

**Comment by ValentinVignal** [comment]
@rrousselGit The `riverpod_lint_flutter_test` is also failing on my local, unless I run `dart pub get` in `riverpod_lint`. I guess this is similar to the issue I have https://github.com/rrousselGit/riverpod/pull/1581#discussion_r952789710

I imagine there is a way to solve this in the Melos configuration?

**Comment by rrousselGit** [comment]
It's fixed on master, but the tests are flacky. I think I'll disable the suite

**Comment by ValentinVignal** [comment]
> It's fixed on master

@rrousselGit I merged the lastest `master` in my branch but `riverpod_lint_flutter_test`'s tests are still failing 🤔

**Comment by rrousselGit** [comment]
Don't worry about it, just tell me when you want this to be merged

File: packages/riverpod_graph/test/integration/counter/counter_test.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import 'dart:io';
+
+import 'package:test/test.dart';
+import 'package:test_process/test_process.dart';
+
+void main() {
+  test('It should log the structure of the addition project', () async {
+    await Process.run('flutter', ['pub', 'get', '../../examples/counter']);
```
[the remark this answers is not in the document]
Comment (ValentinVignal) [reply packages/riverpod_graph/test/integration/counter/counter_test.dart]: In the end I don't use the `examples/`'s projects

**Comment by rrousselGit** [comment]
For info, the CI is fixed on master

**Comment by ValentinVignal** [comment]
Thank you @rrousselGit , this PR should be good to be reviewed now

File: packages/riverpod_graph/test/integration/consumer_widget/golden/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    sdk: flutter
+  flutter_riverpod:
+    path: ../../../../../../packages/flutter_riverpod
+  hooks_riverpod:
+    path: ../../../../../../packages/hooks_riverpod
+  riverpod:
+    path: ../../../../../../packages/riverpod
+
+dev_dependencies:
+  custom_lint: ^0.0.7
```
Comment (rrousselGit) [inline packages/riverpod_graph/test/integration/consumer_widget/golden/pubspec.yaml:23]: Why is this a dependency?

Comment (ValentinVignal) [reply packages/riverpod_graph/test/integration/consumer_widget/golden/pubspec.yaml]: Oops, I removed it in https://github.com/rrousselGit/riverpod/pull/1581/commits/432f572e119383eced0de6435f7408b7f71eaa5a

**Comment by rrousselGit** [comment]
LGTM, thanks a lot!

**Comment by rrousselGit** [comment]
As a next step, fancy looking into supporting the code-generator?

**Comment by ValentinVignal** [comment]
> As a next step, fancy looking into supporting the code-generator?

Sure!

**Pull request merged**